cowprotocol/services
# Remove auction_transaction

# Description
Part of https://github.com/cowprotocol/services/issues/2282 & https://github.com/cowprotocol/services/issues/2275

Exchanges `tx_from/tx_nonce` for `auction_id` which can be directly encoded from the settlement transaction since we shipped colocation. This allows us to remove an entire database table and a bunch of other complicated code.

I'm not yet removing the old columns in case something goes bad with this change.

# Changes


- [x] Removed `auction_transaction` table.
- [x] Redesigned `settlements` table to contain auction id

## How to test
1. db tests and e2e tests cover the new functionality.
2. Locally I've tested the db migration file like this:
- First comment out the V_58 migration file and apply old schema.
- Insert 10 settlements, of which first 5 have `auction_transaction` record (except one settlement (block_number 3), to simulate the existence of settlement from other environment staging/prod)
- Then, manually apply migration script V_58
- Observed the `settlements` table,`auction_id` were populated as expected.
